nexus/tests/integration_tests/authn_http.rs

```diff
@@ -316,7 +315,7 @@ async fn start_whoami_server(
 
 struct WhoamiServerState {
     authn: nexus_db_queries::authn::external::Authenticator<WhoamiServerState>,
-    sessions: Mutex<HashMap<String, FakeSession>>,
+    sessions: Mutex<Vec<FakeSession>>,
```

Comment:
It seems like this could still be a `HashMap`, since you can still iterate over all the entries in a map to find an entry by a different key...not super important, since this is test code, but it might be a bit nicer to just be able to do map lookups when looking up by one of the two key types (ID and token).

Reply:
This felt simpler since neither key is more important than the other for lookups. Not worried about lookup performance.

Reply:
Fair enough, I really don't care either way; the plus IMO was not lookup performance but that the code for one of the two lookups would be simpler and could remain unchanged in this diff. It's definitely not important to me.